### ISSUE TITLE
Add more detail on procedural macros and custom derive

### ISSUE THREAD

**Issue by rylev** [rust-lang/rustc-dev-guide#1160]
The [section on procedural macros](https://github.com/rust-lang/rustc-dev-guide/blob/master/src/macro-expansion.md#procedural-macros) is lacking (especially with regards to custom derive). This should be fleshed out more.

**Comment by linyihai** [comment]
Yes, is there any plan to on the [the section on procedural macros](https://github.com/rust-lang/rustc-dev-guide/blob/master/src/macro-expansion.md#procedural-macros), or any recommend material about procedural macros?
